modules/planning/tasks/optimizers/piecewise_jerk_speed/piecewise_jerk_speed_nonlinear_optimizer.cc

```diff
@@ -149,12 +149,31 @@ Status PiecewiseJerkSpeedNonlinearOptimizer::Process(
   const auto& piecewise_jerk_nonlinear_speed_config =
       config_.piecewise_jerk_nonlinear_speed_config();
 
+  // TODO(Jinyun): evaluate the performance of piecewise linear path data, will
+  // try piecewise polynomial path data
   ptr_interface->set_path(path_data);
 
   // TODO(Jinyun): add speed_limit fitting curve
   //   ptr_interface->set_speed_limit_curve(speed_limit);
 
-  // TODO(Jinyun): add ref_s into optimizer
+  // TODO(Jinyun): evaluate the performance of piecewise linear warm start st
+  // profile, will try piecewise polynomial st profile
+  std::vector<std::vector<double>> warm_start;
+  SpeedData reference_speed_data = *speed_data;
+  for (int i = 0; i < num_of_knots; ++i) {
+    SpeedPoint sp;
+    if (!reference_speed_data.EvaluateByTime(i * delta_t, &sp)) {
+      std::string msg("retriving dp st data for speed optimization failed!");
+      AERROR << msg;
+      speed_data->clear();
+      return Status(ErrorCode::PLANNING_ERROR, msg);
+    }
+    warm_start.emplace_back(
+        std::initializer_list<double>{sp.s(), sp.v(), sp.a()});
```

Comment:
I guess that the `std::initializer_list<double>` is not needed, as `{}` already means an initializer_list.